### PR TITLE
Add new module: cmseq/polymut

### LINKED ISSUE #897: new module: cmseq/polymut
I think it would be good to have a module for [cmseq/polymut](https://github.com/SegataLab/cmseq#polymorphic-rate-over-protein-coding-genes-with-polymutpy)

- [x] This module does not exist yet with the [`nf-core modules list`](https://github.com/nf-core/tools#list-modules) command
- [x] There is no [open pull request](https://github.com/nf-core/modules/pulls) for this module
- [x] There is no [open issue](https://github.com/nf-core/modules/issues) for this module
- [x] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by maxibor**
Add new module to close #897 

The version is for now manually exported (`echo "1.0.4"`) as the tool doesn't have a `--version` flag in its CLI, but a PR is open to add it https://github.com/SegataLab/cmseq/pull/8

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: modules/cmseq/polymut/main.nf [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    script:
+    def prefix = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    def fasta_refid = fasta ? "-c $fasta" : ""
+    """
+    polymut.py \\
+        $options.args \\
+        $fasta_refid \\
+        --gff_file $gff \\
+        $bam > ${prefix}.polymut.txt
```
[the remark this answers is not in the document]
Comment (maxibor) [reply modules/cmseq/polymut/main.nf:35]: I prefer to leave like this so you know where the file is coming from.

File: modules/cmseq/polymut/main.nf [hunk 2]
```diff
@@ -0,0 +1,44 @@
+include { initOptions; saveFiles; getSoftwareName; getProcessName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
```
Comment (jfy133) [inline modules/cmseq/polymut/main.nf:5]:
```suggestion

def VERSION = '4.11'

```

File: modules/cmseq/polymut/main.nf [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    polymut.py \\
+        $options.args \\
+        $sortindex \\
+        $fasta_refid \\
+        --gff_file $gff \\
+        $bam > ${prefix}.polymut.txt
+
+    cat <<-END_VERSIONS > versions.yml
+    ${getProcessName(task.process)}:
+        ${getSoftwareName(task.process)}: \$( echo "1.0.4" )
```
Comment (jfy133) [inline modules/cmseq/polymut/main.nf:41]:
```suggestion
        ${getSoftwareName(task.process)}: \$( echo $VERSION )
```

File: modules/cmseq/polymut/main.nf (hunk 1 above)
[the remark this answers is not in the document]
Comment (jfy133) [reply modules/cmseq/polymut/main.nf:37]: I would also normally lean towards that, but again it's up to pipeline developers to decide this (ultimately). Idea being that you can't assume exactly what they want to do with the tool, I guess (see the freebayes debacle ;) )

https://nf-co.re/developers/adding_modules#naming-conventions

File: modules/cmseq/polymut/meta.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - versions:
+      type: file
+      description: File containing software versions
+      pattern: "versions.yml"
+  - polymut:
+      type: file
+      description: Polymut report in `.txt` format.
+      pattern: "*.polymut.txt"
```
Comment (jfy133) [inline modules/cmseq/polymut/meta.yml:58]:
```suggestion
      pattern: "*.txt"
```

File: tests/modules/cmseq/polymut/test.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    - path: output/cmseq/test.polymut.txt
+      md5sum: fd325c1724ee23d132a9115c64494efc
+
+- name: cmseq polymut test_cmseq_polymut_2
+  command: nextflow run tests/modules/cmseq/polymut -entry test_cmseq_polymut_2 -c tests/config/nextflow.config
+  tags:
+    - cmseq
+    - cmseq/polymut
+  files:
+    - path: output/cmseq/test.polymut.txt
```
Comment (jfy133) [inline tests/modules/cmseq/polymut/test.yml:16]:
```suggestion
    - path: output/cmseq/test.txt
```

File: tests/modules/cmseq/polymut/test.yml [hunk 6]
```diff
@@ -0,0 +1,26 @@
+- name: cmseq polymut test_cmseq_polymut_1
+  command: nextflow run tests/modules/cmseq/polymut -entry test_cmseq_polymut_1 -c tests/config/nextflow.config
+  tags:
+    - cmseq
+    - cmseq/polymut
+  files:
+    - path: output/cmseq/test.polymut.txt
```
Comment (jfy133) [inline tests/modules/cmseq/polymut/test.yml:7]:
```suggestion
    - path: output/cmseq/test.txt
```

File: tests/modules/cmseq/polymut/test.yml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+      md5sum: fd325c1724ee23d132a9115c64494efc
+
+- name: cmseq polymut test_cmseq_polymut_3
+  command: nextflow run tests/modules/cmseq/polymut -entry test_cmseq_polymut_3 -c tests/config/nextflow.config
+  tags:
+    - cmseq
+    - cmseq/polymut
+  files:
+    - path: output/cmseq/test.polymut.txt
```
Comment (jfy133) [inline tests/modules/cmseq/polymut/test.yml:25]:
```suggestion
    - path: output/cmseq/test.txt
```

**Review by jfy133 — COMMENTED** [review]
Slight tweaks based on guidelines, and suggestion for following standard for non-reporting versions

**Review by jfy133 — APPROVED** [review]
Just missing the standarded way of reporting versions (when the tool doesn't do it itself) as in the last point [here](https://nf-co.re/developers/adding_modules#general), and you're good to go :+1:

**Pull request merged**